content/guide/property-system.md

```diff
@@ -0,0 +1,315 @@
+---
+title: Property
+---
+
+The property system in NativeScript is a way to expose a native property when you are exposing a native UI component.
+
+NativeScript provides a basic property system via the `Property` class, which is a simple wrapper around the [Object.defineProperty](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) with some additional callbacks like `valueChange`, `valueConverter` and `equalityComparer`.
+
+## Using Property
+
+### Adding a new property to a View instance
+Add new properties by creating a new `Property` with desired [options](#propertyoptions-interface).
+
+```ts
+class MyButtonBase extends Button{}
+export const textProperty = new Property<MyButtonBase, string>({
+    name: "text",
+    defaultValue: "",
+    affectsLayout: true
+})
+```
+Then register the property with the desired class:
+
+```ts
+textProperty.register(MyButtonBase);
+```
+To apply the JS value to the exposed native view, call the `setNative` method on the property instance.
+
+```ts
+[textProperty.setNative](value: string) {
+    this.nativeView.text = value
+}
+```
+For a complete real world example about exposing a native UI Component to NativeScript developers, and how the Property system is used, read the [How to create a NativeScript plugin that uses native iOS & Android views (Part 1) - Label Marquee!](https://blog.nativescript.org/create-a-custom-view-plugin-marquee-label/) article.
```

Comment:
To gain a comprehensive understanding of how the Property system is employed in NativeScript to expose a native UI component, I recommend reading the informative article titled [How to create a NativeScript plugin that uses native iOS & Android views (Part 1) - Label Marquee!](https://blog.nativescript.org/create-a-custom-view-plugin-marquee-label/). This article provides a practical real-world example that demonstrates the process of developing a NativeScript plugin while utilizing native iOS and Android views.